requirements/README.md

```diff
@@ -1,23 +1,49 @@
+At the end of both `.in` files, pip and setuptools need to have their versions pinned.
+After producing the `.txt` files, the versions need to be manually added back at the end.
 
-At the end of `requirements/requirements.in`, pip and setuptools need to have their versions pinned.
+Run these commands from the root of the awx repo. This will produce python 3 requirements files.
 
 ```
-python3 -m venvpip install pip-tools /buildit
+python3 -m venv /buildit
 source /buildit/bin/activate
 pip install pip-tools
 pip install pip --upgrade
 
-pip-compile requirements/requirements.in > requirements/requirements.txt
-pip-compile requirements/requirements_ansible.in > requirements/requirements_ansible.txt
+pip-compile --output-file requirements/requirements.txt requirements/requirements.in
+pip-compile --output-file requirements/requirements_ansible_py3.txt requirements/requirements_ansible.in
 ```
 
-## Known Issues
+The Ansible venv requirements file needs to start with the python 2 version
+as a base. Then we can run the tool again to get the python 3 version.
+Consult the output of the `diff` command and add a conditional switch in those cases.
 
-* Remove the `-e` from packages of the form `-e git+https://github.com...` in the generated `.txt`. Failure to do so will result in a "bad" RPM and DEB due to the `pip install` laying down a symbolic link with an absolute path from the virtualenv to the git repository that will differ from when the RPM and DEB are build to when the RPM and DEB are installed on a machine. By removing the `-e` the symbolic egg link will not be created and all is well.
+```
+virtualenv -p python2 /buildit_py2
+source /buildit_py2/bin/activate
+pip install pip-tools
+pip install pip --upgrade
+
+pip-compile --output-file requirements/requirements_ansible.txt requirements/requirements_ansible.in
+diff requirements/requirements_ansible_py3.txt requirements/requirements_ansible.txt
+rm requirements/requirements_ansible_py3.txt
+```
+
+Python 3 exceptions should be added to relevant `requirements_ansible.txt` lines
+after version numbers with the syntax of `; python_version < '3'`.
+
+## Licenses and Source Files
+
+If any library has a change to its license with the upgrade, then the license for that library
+inside of `docs/licenses` needs to be updated.
+
+For libraries that have source distribution requirements (LGPL as an example),
+a tarball of the library is kept along with the license.
+To download the PyPI tarball, you can run this command:
+
+```
+pip download <pypi library name> -d docs/licenses/ --no-binary :all: --no-deps
```

Comment:
Man, thanks for putting in the legwork to get this process updated.  It needed some love.